File: WolfDen/Game/Behaviours/CombatBehaviour/CannotAttackCombatBehaviourTemplate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using WolfDen.Game;
using WolfDen.Game.Interfaces;

namespace WolfDen.Game.Behaviours.CombatBehaviour
{
    public class CannotAttackCombatBehaviourTemplate : NormalCombatBehaviourTemplate
    {
        public override void AddBehaviour(Unit unit)
        {
            // If unit has move behaviour make sure it was added before this behaviour.
            var moveBehaviour = unit.Behaviours.Values.OfType<IMoveBehaviour>().FirstOrDefault();
            var combatBehaviour = new CannotAttackCombatBehaviour(unit.Game, unit, MaxHealth, UnitType, moveBehaviour, DamageChart);
            combatBehaviour.Id = unit.Id + Class;
            unit.Behaviours.Add(combatBehaviour.Id, combatBehaviour);
        }
    }
}

File: WolfDen/wwwroot/ts/Behaviours/CombatBehaviour/CannotAttackCombatBehaviourTemplate.ts
/// <reference path="../../Unit.ts" />
/// <reference path="NormalCombatBehaviourTemplate.ts" />

namespace TurnBasedBoardGame {
    export class CannotAttackCombatBehaviourTemplate extends NormalCombatBehaviourTemplate {

        AddBehaviour(unit: Unit): void {
            let self = this;
            let combatBehaviour = new CannotAttackCombatBehaviour();
            Object.assign(combatBehaviour, self);
            combatBehaviour.Unit = unit;
            combatBehaviour.Game = unit.Game;
            combatBehaviour.CurrentHealth = self.MaxHealth;
            combatBehaviour.Id = unit.Id + self.Class;
            combatBehaviour.Initialize();
            
            unit.Behaviours[combatBehaviour.Id] = combatBehaviour;
        }
    }
}